app/styles/atoms/progress-bar.less

```diff
@@ -0,0 +1,42 @@
+.oss-progress-bar--wrapper {
+  display: flex;
+  width: 100%;
+  justify-content: space-between;
+  align-items: center;
+  span {
+    font-weight: 200;
+    margin: 0 var(--spacing-px-6);
+  }
+}
+
+.oss-progress-bar {
+  width: 100%;
+  border-radius: var(--border-radius-sm);
+  background-color: #e5e5e5;
+  margin-bottom: 0;
+  overflow: hidden;
+
+  .oss-progress-bar--inner {
+    width: 0;
+    animation: oss-progress-bar-animation 1s forwards;
+    border-radius: var(--border-radius-sm) 0 0 var(--border-radius-sm);
+    transition: width 0.6s ease;
+    height: 10px;
+
+    &--small {
+      height: 4px;
+    }
+
+    &--primary {
+      background-color: var(--color-primary-500);
+    }
+
+    &--warning {
+      background-color: var(--color-warning-500);
+    }
+
+    &--success {
```

Comment:
why not put the `--modifier` on the parent class instead? (e.g: `.oss-progress-bar.oss-progress-bar--success`)

Reply:
The inner takes the class because I don't want to override the one in the parent. Parent takes  --color-gray-200, inner takes modifier . I made it bem compliant though